babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -604,19 +605,33 @@ SqlCreate SqlCreateTable(Span s, SqlCreateSpecifier createSpecifier) :
     |
         { query = null; }
     )
-    (
-       index = SqlCreateTableIndex(s)
-       {
+    [
+        index = SqlCreateTableIndex(s)
+        {
            if (index instanceof SqlPrimaryIndex) {
                primaryIndex = (SqlPrimaryIndex) index;
            }
-       }
-    )*
+           else{
+               indices.add(index);
+           }
+        }
+        (
```

Comment:
Is there a better way of ensuring that index declarations are comma separated (without the first declaration needing to be preceded by a comma)?

Reply:
One thing that could simplify the parsing logic is to just put all indices, regardless of if they're primary or secondary, into the "indices" list that is passed into the SqlCreateTable constructor. Within the SqlCreateTable java code, you could enforce any constraints that you need to.

Then, you could break this portion of the parsing logic out into a helper function and simplify into something like:

void ExtractIndex(List<SqlIndex> indices):
{
}
{
   index = SqlCreateTableIndex(s)
   {
     indices.add(index);
   }
}

SqlCreate SqlCreateTable(Span s, SqlCreateSpecifier createSpecifier) :
{
...
  List<SqlIndex> indices = new ArrayList<SqlIndex>();
...
}
{
...
ExtractIndex(indices)
( COMMA ExtractIndex(indices) )*
...
}

Reply:
As per our conversation, I have adapted the parsing logic to add all indices to the list and then extract the primary ones (to minimize duplicated code), but have refrained from doing this inside the SqlCreateTable class so as to avoid reworking existing constructors.